File: OOPDraw/Shape.cs
using Nakov.TurtleGraphics;
using System.Drawing;
using System;

namespace OOPDraw
{
    public abstract class Shape
    {
        //Properties
        public float OriginX { get;  set; }
        public float OriginY { get;  set; }
        public float LineWidth { get; private set; }

        public Shape(float centreX, float centreY)
        {
            OriginX = centreX;
            OriginY = centreY;
        }

        //Abstract methods -  to be implemeted in sub-types
        public abstract void Draw();

        public void ResizeTo(int x, int y)
        {
            Resize(Math.Abs(x - OriginX), Math.Abs(y - OriginY));
        }

        public abstract void Resize(float x, float y);

        //Concrete methods
        public virtual void MoveTo(float x, float y)
        {
            OriginX = x;
            OriginY = y;
        }

        public virtual void MoveBy(float x, float y)
        {
            OriginX += x;
            OriginY += y;
        }

        public virtual void Select()
        {
            LineWidth = 4;
        }

        public virtual void Unselect()
        {
            LineWidth = 2;
        }

        /// <summary>
        /// Reset's the position, angle, pen size and colour to the specified properties for this shape
        /// </summary>
        protected void ResetTurtle()
        {
            Turtle.Angle = 0;
            Turtle.PenColor = Color.Black;
            Turtle.PenSize = LineWidth;
            Turtle.X = OriginX;
            Turtle.Y = OriginY;
        }
    }
}

File: OOPDraw VB/OOPDraw/Shape.vb
Imports Nakov.TurtleGraphics

Public MustInherit Class Shape
    Protected Property XOrigin As Single
    Protected Property YOrigin As Single
    Private Property LineWidth As Single

    Public Sub New(xOrigin As Single, yOrigin As Single)
        Me.XOrigin = xOrigin
        Me.YOrigin = yOrigin
    End Sub

    Public Overridable Sub SelectShape()
        LineWidth = 4
    End Sub

    Public Overridable Sub UnselectShape()
        LineWidth = 2
    End Sub

    Public MustOverride Sub Draw()

    Public Overridable Sub MoveTo(x As Single, y As Single)
        XOrigin = x
        YOrigin = y
    End Sub

    Public Sub MoveBy(x As Single, y As Single)
        XOrigin += x
        YOrigin += y
    End Sub

    Public Sub ResizeAbsolute(turtleX As Single, turtleY As Single)
        Resize(Math.Abs(turtleX - XOrigin), Math.Abs(turtleY - YOrigin))
    End Sub

    Public MustOverride Sub Resize(x As Single, y As Single)

    Protected Sub ResetTurtle()
        Turtle.ShowTurtle = False
        Turtle.PenSize = LineWidth
        Turtle.Angle = 0 'Always start from North
        Turtle.X = XOrigin
        Turtle.Y = YOrigin
    End Sub
End Class
